packages/common/lib/xml-ns-key.js

```diff
@@ -0,0 +1,38 @@
+// The xml parser takes care of validating the attribute name.
+// If the user started the attribute name with "xmlns:" we can assume that
+// they meant for it to be an xml namespace attribute.
+// xmlns attributes explicitly can't contain ":" after the "xmlns:" part.
+const namespaceRegex = /^xmlns(?<prefixWithColon>:(?<prefix>[^:]*))?$/;
+
+function isXMLNamespaceKey(key, includeEmptyPrefix) {
+  if (typeof key !== "string") {
+    return false;
+  }
+  const matchArr = key.match(namespaceRegex);
+  if (matchArr === null) {
+    return false;
+  }
+  if (includeEmptyPrefix === true) {
+    return true;
+  }
+  /* istanbul ignore next - defensive coding */
+  const groups = matchArr.groups || {};
```

Comment:
I don't think we need the ` || {}` in JavaScript.

Because we know that in Node 10.0+ named capture groups are supported and that
the match succeed if we reached this line.

In TypeScript we used ` || {}` to calm to compiler as the `.groups` property is defined as optional. so its not really defensive coding, we can **prove** it can't reach this branch so we don't actually need the branch.

Reply:
ok, I'll remove it